pkg/apis/provisioning/v1alpha5/labels.go

```diff
@@ -53,10 +53,24 @@ var (
 		v1.LabelNamespaceSuffixNode,
 	)
 
-	// WellKnownLabels are labels that belong to the RestrictedLabelDomains but allowed.
-	// Karpenter is aware of these labels, and they can be used to further narrow down
-	// the range of the corresponding values by either provisioner or pods.
-	WellKnownLabels = sets.NewString(
+	// WellKnownLabelsForProvisioner are labels that belong to the RestrictedLabelDomains but
+	// allowed in provisioner. Karpenter is aware of these labels.
+	WellKnownLabelsForProvisioner = sets.NewString(
```

Comment:
Sorry if this is a bit confusing, but WellKnownLabels for provisioner is identical to WellKnownLabels except for `ProvisionerNameLabelKey` However, the `WellKnownLabels` public variable is actually mutated by cloud providers to add provider specific labels, so we need to be a little careful about duplicating the slices. 

Instead, I think we can fix this by adding another validation (https://github.com/aws/karpenter/blob/main/pkg/apis/provisioning/v1alpha5/provisioner_validation.go#L91) to check that the provisioner name label isn't set. All other labels between pods/provisioners will be identical, so we can just rely on a single WellKnownLabels variable.

Reply:
You should be able to test this by creating a provisioner with a provider specific label like `karpenter.k8s.aws/instance-family`.

Reply:
@ellistarn Ah, thank you for following up! I agree with you 🙏 
> However, the WellKnownLabels public variable is actually mutated by cloud providers to add provider specific labels, so we need to be a little careful about duplicating the slices.

Also, many thanks for your helps in https://github.com/aws/karpenter/pull/2328, I fixed this PR based on the pull request (and marked the commit with co-authored with you.)